diff --git a/work/acpi_call-1.1.0/acpi_call.c b/work/acpi_call-1.1.0/acpi_call.c
index 3025d97..602d71a 100644
--- acpi_call-1.1.0/acpi_call.c
+++ acpi_call-1.1.0/acpi_call.c
@@ -5,9 +5,12 @@
 #include <linux/version.h>
 #include <linux/proc_fs.h>
 #include <linux/slab.h>
-#include <asm/uaccess.h>
 #include <linux/uaccess.h>
-#include <linux/acpi.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0)
+    #include <linux/acpi.h>
+#else
+    #include <acpi/acpi.h>
+#endif
 
 MODULE_LICENSE("GPL");
 
